File: src/main/java/com/owncloud/android/ui/trashbin/TrashbinContract.java
/*
 * Nextcloud Android client application
 *
 * @author Tobias Kaminsky
 * Copyright (C) 2018 Tobias Kaminsky
 * Copyright (C) 2018 Nextcloud GmbH.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
package com.owncloud.android.ui.trashbin;

import com.owncloud.android.lib.resources.trashbin.model.TrashbinFile;

import java.util.List;

/**
 * Contract between view (TrashbinActivity) and presenter (TrashbinPresenter)
 */
public interface TrashbinContract {

    interface View {
        void showTrashbinFolder(List<Object> trashbinFiles);

        void showSnackbarError(int message, TrashbinFile file);

        void showError(int message);

        void removeFile(TrashbinFile file);

        void removeAllFiles();

        void close();

        void setDrawerIndicatorEnabled(boolean bool);
    }

    interface Presenter {

        boolean isRoot();

        void loadFolder();

        void navigateUp();

        void enterFolder(String folder);

        void restoreTrashbinFile(TrashbinFile file);

        void removeTrashbinFile(TrashbinFile file);

        void emptyTrashbin();
    }
}

File: app/src/main/java/com/owncloud/android/ui/trashbin/TrashbinContract.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2024 TSI-mc <surinder.kumar@t-systems.com>
 * SPDX-FileCopyrightText: 2018 Tobias Kaminsky <tobias@kaminsky.me>
 * SPDX-FileCopyrightText: 2018 Nextcloud GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.ui.trashbin

import com.owncloud.android.lib.resources.trashbin.model.TrashbinFile

/**
 * Contract between view (TrashbinActivity) and presenter (TrashbinPresenter)
 */
interface TrashbinContract {
    interface View {
        fun showTrashbinFolder(trashbinFiles: List<TrashbinFile?>?)
        fun showSnackbarError(message: Int, file: TrashbinFile?)
        fun showError(message: Int)
        fun removeFile(file: TrashbinFile?)
        fun removeAllFiles()
        fun close()
        fun atRoot(isRoot: Boolean)
    }

    interface Presenter {
        val isRoot: Boolean
        fun loadFolder(onCompleted: () -> Unit = {}, onError: () -> Unit = {})
        fun navigateUp()
        fun enterFolder(folder: String?)
        fun restoreTrashbinFile(files: Collection<TrashbinFile?>)
        fun removeTrashbinFile(files: Collection<TrashbinFile?>)
        fun emptyTrashbin()
    }
}
